core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -286,19 +286,19 @@ class BrokerTopicMetrics(name: Option[String], remoteStorageEnabled: Boolean = f
   case class GaugeWrapper(metricType: String) {
     @volatile private var gaugeObject: Gauge[Long] = _
     final private val gaugeLock = new Object
-    final val aggregatedMetric = new AggregatedMetric()
+    final private val metric = new AggregatedMetric()
 
-    def gauge(): Gauge[Long] = gaugeLock synchronized {
+    def gauge(): AggregatedMetric = gaugeLock synchronized {
       if (gaugeObject == null) {
-        gaugeObject = metricsGroup.newGauge(metricType, () => aggregatedMetric.value(), tags)
+        gaugeObject = metricsGroup.newGauge(metricType, () => metric.value(), tags)
       }
-      return gaugeObject
+      return metric
     }
 
     def close(): Unit = gaugeLock synchronized {
```

Comment:
it seems the lock is used to make sure we don't call removeMetric/newGauge repeatedly, so we can do a bit refactor for it. WDYT?
```scala
  case class GaugeWrapper(metricType: String) {
    private final val removed = new AtomicBoolean(false)
    // The map to store:
    //   - per-partition value for topic-level metrics. The key will be the partition number
    //   - per-topic value for broker-level metrics. The key will be the topic name
    private val metricValues = new ConcurrentHashMap[String, Long]()

    def setValue(key: String, value: Long): Unit = {
      newGaugeIfNeed()
      metricValues.put(key, value)
    }

    def removeKey(key: String): Unit = {
      newGaugeIfNeed()
      metricValues.remove(key)
    }

    def close(): Unit = if (removed.compareAndSet(true, false)) {
      metricsGroup.removeMetric(metricType, tags)
      metricValues.clear()
    }

    private def newGaugeIfNeed(): Unit = if (removed.compareAndSet(false, true)) {
      metricsGroup.newGauge(metricType, () => metricValues.values().stream().mapToLong(v => v).sum(), tags)
    }

    newGaugeIfNeed()
  }
```